x/xcvm/keeper/transfer_intent.go

```diff
@@ -0,0 +1,448 @@
+	}
+	coins := sdk.NewCoins(transferIntent.Bounty)
+
+	// Release bounty to original sender
+	if err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, accAddress, coins); err != nil {
+		return fmt.Errorf("release unclaimed bounty to original sender: %v", err)
+	}
+
+	// Remove transfer intent from store
+	kvStore.Delete(types.GetPendingTransferIntentKeyById(msg.IntentId))
+
+	return nil
+}
+
+// verifyReceiptSignature verifies that the solver signed over its proof using the same Ethereum account with which it fulfills the intent
+func verifyReceiptSignature(solverPublicKey *ecdsa.PublicKey, txReceiptHash []byte, blockHash common.Hash, receiptSig []byte) error {
+	encPublicKey := crypto.FromECDSAPub(solverPublicKey)
+	receiptDataHash := crypto.Keccak256(append(txReceiptHash, blockHash.Bytes()...))
+
+	// Remove the recovery id from the receipt signature before verifying
+	if !crypto.VerifySignature(encPublicKey, receiptDataHash, receiptSig[:64]) {
+		return types.ErrInvalidReceiptSignature
+	}
+	return nil
+}
+
+// verifyReceiptProof verifies that the receipt is included in the block header's receipt trie
+func verifyReceiptProof(blockHeader gethtypes.Header, txReceiptHash []byte, receiptProofBz []byte) error {
+	var receiptProof types.ReceiptProof
+	if err := receiptProof.Unmarshal(receiptProofBz); err != nil {
+		return fmt.Errorf("unmarshal receipt proof: %v", err)
+	}
+
+	receiptsRoot := blockHeader.ReceiptHash
+	if _, err := trie.VerifyProof(receiptsRoot, txReceiptHash, receiptProof); err != nil {
+		return fmt.Errorf("verify proof: %v", err)
+	}
+
+	return nil
+}
+
+// getTxReceiptHash returns the hash of the transaction receipt
+func getTxReceiptHash(txReceipt gethtypes.Receipt) ([]byte, error) {
+	//Get binary representation of txReceipt rlp encoding
+	txReceiptBz, err := txReceipt.MarshalBinary()
+	if err != nil {
+		return nil, fmt.Errorf("get tx receipt hash: %v", err)
+	}
+	txReceiptHash := crypto.Keccak256(txReceiptBz)
+	return txReceiptHash, nil
+}
+
+// verifyReceiptUniqueness verifies that the receipt has not been processed before
+func verifyReceiptUniqueness(store store.KVStore, txReceiptHash []byte, blockHash common.Hash) error {
+	receiptKey := types.GetUsedReceiptKey(txReceiptHash, blockHash)
+	if store.Has(receiptKey) {
+		return types.ErrReceiptAlreadyProcessed
+	}
+
+	store.Set(receiptKey, []byte{0x01})
+	return nil
+}
+
+// verifyPreviousBeaconBlockHeaders verifies that the provided previous beacon block headers are valid and in the correct order
+func verifyPreviousBeaconBlockHeaders(beaconBlockHeaders []*types.BeaconBlockHeader) error {
+	intentBeaconBlockHeader := *beaconBlockHeaders[0]
+	headerHash, err := intentBeaconBlockHeader.Hash()
+	if err != nil {
+		return fmt.Errorf("hash intent beacon block header: %v", err)
+	}
+
+	var currentParentRoot [32]byte
+	for _, header := range beaconBlockHeaders[1:] {
+		copy(currentParentRoot[:], header.GetParentRoot())
```

Comment:
nit: no need to copy. `parentRoot := header.GetParentRoot()` preferred unless I'm missing something

Reply:
The reason I ended up copying the root was because of a mismatched types error while comparing the parent hash with the header hash `mismatched types []byte and [32]byte`. I'll opt to change the func to get the hash of the beacon block header to return a slice rather than an array so that I don't need to copy here.

Reply:
After looking at making the change it seems like I'll need an external package to compare slices, do you have any ideas on how to avoid copying the parent hash into an array?